service_discovery/dto/payment_method.go

```diff
@@ -17,9 +17,49 @@
 
 package dto
 
-import "github.com/mysteriumnetwork/node/money"
+import (
+	"encoding/json"
 
+	"github.com/mysteriumnetwork/node/money"
+)
+
+// PaymentMethod is a marker interface for all types of payment methods
 type PaymentMethod interface {
 	// Service price per unit of metering
 	GetPrice() money.Money
 }
+
+// UnsupportedPaymentMethod represents payment method which is uknown to node (i.e. not registered)
+type UnsupportedPaymentMethod struct {
+}
+
+// GetPrice always panics for UnsupportedPaymentMethod and should not be called
+func (UnsupportedPaymentMethod) GetPrice() money.Money {
+	//this should never be called
+	panic("not supported")
+}
+
+var _ PaymentMethod = UnsupportedPaymentMethod{}
+
+// PaymentMethodUnserializer is function type which takes raw json message and returns deserialized payment method
+type PaymentMethodUnserializer func(*json.RawMessage) (PaymentMethod, error)
+
+// service payment method unserializer registry
+var paymentMethodMap = make(map[string]PaymentMethodUnserializer, 0)
+
+// RegisterPaymentMethodUnserializer registers unserializer for specified payment method type
+func RegisterPaymentMethodUnserializer(paymentMethod string, unserializer func(*json.RawMessage) (PaymentMethod, error)) {
```

Comment:
Should it return error? Don't see its usage.

Reply:
That's the unserializer function which returns error. Register function doesn't return anything